One test failed

```
  FAILED tests/utils/test_helpers.py::TestHelpers::test_build_airflow_url_with_query
```

URL: https://github.com/apache/airflow/pull/14990/checks?check_run_id=2293633953#step:6:11359

--- a/airflow/utils/helpers.py
+++ b/airflow/utils/helpers.py
@@ -24,6 +24,7 @@ from itertools import filterfalse, tee
 from typing import Any, Callable, Dict, Generator, Iterable, List, Optional, TypeVar
 from urllib import parse
 
+from flask import url_for
 from jinja2 import Template
 
 from airflow.configuration import conf
@@ -213,4 +214,5 @@ def build_airflow_url_with_query(query: Dict[str, Any]) -> str:
     'http://0.0.0.0:8000/base/graph?dag_id=my-task&root=&execution_date=2020-10-27T10%3A59%3A25.615587
     """
     view = conf.get('webserver', 'dag_default_view').lower()
-    return f"/{view}?{parse.urlencode(query)}"
+    url = url_for(f"Airflow.{view}")
+    return f"{url}?{parse.urlencode(query)}"

--- a/tests/utils/test_helpers.py
+++ b/tests/utils/test_helpers.py
@@ -18,6 +18,7 @@
 
 import unittest
 from datetime import datetime
+from unittest import mock
 
 import pytest
 
@@ -140,10 +141,18 @@ class TestHelpers(unittest.TestCase):
 
     @conf_vars(
         {
-            ("webserver", "dag_default_view"): "custom",
+            ("webserver", "dag_default_view"): "graph",
         }
     )
     def test_build_airflow_url_with_query(self):
+        """
+        Test query generated with dag_id and params
+        """
         query = {"dag_id": "test_dag", "param": "key/to.encode"}
-        url = build_airflow_url_with_query(query)
-        assert url == "/custom?dag_id=test_dag&param=key%2Fto.encode"
+        expected_url = "/graph?dag_id=test_dag&param=key%2Fto.encode"
+
+        from airflow.www.app import cached_app
+
+        with cached_app(testing=True).test_request_context():
+            assert build_airflow_url_with_query(query) == expected_url
+

--- a/tests/utils/test_helpers.py
+++ b/tests/utils/test_helpers.py
@@ -155,4 +155,3 @@ class TestHelpers(unittest.TestCase):
 
         with cached_app(testing=True).test_request_context():
             assert build_airflow_url_with_query(query) == expected_url
-

--- a/tests/utils/test_helpers.py
+++ b/tests/utils/test_helpers.py
@@ -18,7 +18,6 @@
 
 import unittest
 from datetime import datetime
-from unittest import mock
 
 import pytest
 

